### PR TITLE
CSS parsing, GPU rendering, URL parsing

### PR DESCRIPTION
**Pull request by alx741**
- CSS parsing detail added
- GPU rendering process (no hardware internals yet)
- URL parsing (URL parts)
- Virtual Keyboard case added

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -118,8 +118,31 @@ appropiate font symbol in the appropiate focused field.
 Is it a URL or a search term?
 -----------------------------
 
-Parse URL...
-------------
+Parse URL
+---------
+
+* The browser knows what to do based on the iformation provided by the URL,
```
Comment (phildini) [inline README.rst]: information

File: README.rst [hunk 2]
```diff
@@ -118,8 +118,31 @@ appropiate font symbol in the appropiate focused field.
...
+  where:
+
+  *``http://www.google.com``*
+
+  - ``http://``
+    Tells the browser that it needs to communicate using the 'Hiper Text
```
Comment (phildini) [inline README.rst]: Hyper

File: README.rst [hunk 3]
```diff
@@ -462,6 +489,15 @@ Page Rendering
 GPU Rendering
 -------------
 
+* During rendering process the graphical computing layers can use general
+  purpose ``CPU`` or the grhapical processor ``GPU`` as well.
```
Comment (phildini) [inline README.rst]: graphical

File: README.rst [hunk 4]
```diff
@@ -462,6 +489,15 @@ Page Rendering
...
+
+* When using ``GPU`` for graphical rendering computations the graphical
+  software layers split the task into multple pieces, so it can take advantage
+  of ``GPU`` massive parallelism for float point calculations requiered for
```
Comment (phildini) [inline README.rst]: required

**Comment by alx741** [comment]
I guess a bunch if issues need to be closed:
#75  #43 #10

File: README.rst [hunk 5]
```diff
@@ -33,22 +33,48 @@ The keyboard controller then encodes the keycode for transport to the computer.
...
+  device driver.  The value of the key is then passed into the operating
+  system's hardware abstraction layer.
+
+*In the case of Virtual Keyboard (as in touch screen devices):*
+
+- In modern capacitive touch screens when the user puts his finger on the
+  screen a tiny amount of current from the electrostatic field of the
+  conductive layer gets transferred to the finger completing the circuit,
+  creating a voltage dropping at that point on the screen that the
+  ``screen controller`` which rises an interrupt reporting the coordinate of
```
Comment (phildini) [inline README.rst]: s/rises/raises?

File: README.rst [hunk 6]
```diff
@@ -33,22 +33,48 @@ The keyboard controller then encodes the keycode for transport to the computer.
...
+  the 'click'.
+
+- Then the mobile OS notify the current focused application of a click event in
```
Comment (phildini) [inline README.rst]: notifies

File: README.rst [hunk 7]
```diff
@@ -33,22 +33,48 @@ The keyboard controller then encodes the keycode for transport to the computer.
...
+  one of its GUI elements (which now is the virtual keyboard application
+  buttons).
+
+- The virtual keyboard can now rises a software interrupt for sending a
+  'key pressed' message back to the OS.
+
+- Which in turn notify the current focused application of a 'key pressed' event
```
Comment (phildini) [inline README.rst]: notifies

File: README.rst [hunk 8]
```diff
@@ -118,8 +144,31 @@ appropiate font symbol in the appropiate focused field.
...
+---------
+
+* The browser knows what to do based on the information provided by the URL,
+  where:
+
+  *``http://www.google.com``*
+
+  - ``http://``
+    Tells the browser that it needs to communicate using the 'Hyper Text
+    Trasnfer Protocol' with the target machine.
```
Comment (phildini) [inline README.rst]: Transfer

**Comment by phildini** [comment]
Reading this carefully, I like the sections on the USB/digital keyboards.

I'm hesitant to approve the section about URL parsing, because it uses terminology that is non-standard or incorrect in some cases. If you were pull out that bit, I would approve the rest.

Thanks for contributing!

**Comment by alx741** [comment]
Sorry about that, not native english speaker + bad spell cheker. Hope it's ok now

File: README.rst [hunk 9]
```diff
@@ -115,11 +142,32 @@ sends the character to the ``window manager`` (DWM, metacity, i3, etc), so the
...
+* The browser has now the following information contained in the URL (Uniform
+  Resource Locator):
+
+    - ``Protocol``  "http"
+        Use 'Hyper Text Transfer Protocol'
+
+    - ``Domain name``  "www.google.com"
+        Connect to host 'www.google.com'
+
+    - ``Web service type``  ".com"
```
Comment (phildini) [inline README.rst]: I would remove these two lines... the information is contained in 155-156, and .com is just a TLD. While it's theoretically supposed to be for 'commercial' services, that's not really respected these days, any more than .ly is only for websites hosted in Lybia.

You could add a note to the 'domain name' section saying that .com stands for Commercial.

**Comment by alx741** [comment]
I've removed that, so i can improve it latter when i've figured out a better way

**Pull request merged**